rust-src/concordium_base/src/common/cbor/primitives.rs

```diff
@@ -308,6 +324,29 @@ impl CborSerialize for MapKey {
     }
 }
 
+fn decode_unsigned_bignum_to_u64<C: CborDecoder>(mut decoder: C) -> CborSerializationResult<u64> {
+    decoder.decode_tag_expect(UNSIGNED_BIGNUM_TAG)?;
+    decode_ne_bytes_to_u64(&decoder.decode_bytes()?)
+}
+
+fn decode_negative_bignum_to_u64<C: CborDecoder>(mut decoder: C) -> CborSerializationResult<u64> {
+    decoder.decode_tag_expect(NEGATIVE_BIGNUM_TAG)?;
+    decode_ne_bytes_to_u64(&decoder.decode_bytes()?)
+}
+
+fn decode_ne_bytes_to_u64(bytes: &[u8]) -> CborSerializationResult<u64> {
+    let bytes_outside_range = &bytes[..bytes.len() - bytes.len().min(8)];
+    if bytes_outside_range.iter().copied().any(|byte| byte != 0) {
+        return Err(anyhow!("bignum out of u64 range").into());
+    }
+
+    let bytes_in_range = &bytes[bytes.len() - bytes.len().min(8)..];
+    let mut u64bytes = [0u8; 8];
+    u64bytes[8 - bytes_in_range.len()..].copy_from_slice(bytes_in_range);
+
+    Ok(u64::from_be_bytes(u64bytes))
```

Comment:
A lot is happening in these few lines :sweat_smile: 
Some comments to help the reader with the reasoning would be great